include/revng/Pipes/CompileModule.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pipeline/ContainerSet.h"
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/Contract.h"
+#include "revng/Pipeline/GenericLLVMPipe.h"
+#include "revng/Pipeline/LLVMContainer.h"
+#include "revng/Pipeline/LLVMGlobalKindBase.h"
+#include "revng/Pipeline/Target.h"
+#include "revng/Pipes/IsolatedKind.h"
+#include "revng/Pipes/RevngPipes.h"
+#include "revng/Pipes/RootKind.h"
+#include "revng/Pipes/TemporaryFile.h"
+
+namespace pipeline {
+
+class CompileModulePipe {
+public:
+  static constexpr auto Name = "Compile Module Pipe";
+  std::array<ContractGroup, 1> getContract() const {
+    return { ContractGroup(Root,
+                           Exactness::DerivedFrom,
+                           0,
+                           Object,
+                           1,
+                           InputPreservation::Preserve) };
```

Comment:
So compiling a LLVM module to an object file is not destructive?

Reply:
Try to `verify` and `dump` when you get the chance. If it's destructive, it's a subtle bug.